### DISCUSSION TITLE
type checking for the project?

### DISCUSSION THREAD

**Discussion by swamper123** [FreeOpcUa/opcua-asyncio#459] [General]
I wanted to ask, what your opinion is about Type checking in the project.
Should we strive to add type annotations in future code? To add a type checker (e.g. like mypy) to our actions?

My opinion is, that this may help us to be more clear in our code and being able to express excatly what kind of e.g. parameter/s should be expected. I don't want to rejig the library in one PR (since that would be massive work), I think that it would be best (not mandatory) practice for future PRs. But this is just my opinion on that topic.

**Comment by oroulet** [comment]
We have already started adding typing several places. Found a few one directly wrong yesterday btw
